src/iostream.c

```diff
@@ -593,14 +593,14 @@ Obj FuncCLOSE_PTY_IOSTREAM(Obj self, Obj stream)
 {
     UInt pty = HashLockStreamIfAvailable(stream);
 
-    PtyIOStreams[pty].inuse = 0;
-
     /* Close down the child */
     int status;
     int retcode = close(PtyIOStreams[pty].ptyFD);
     if (retcode)
         Pr("Strange close return code %d\n", retcode, 0);
     kill(PtyIOStreams[pty].childPID, SIGTERM);
+    // GAP (or another library) might wait on this PID before
+    // we handle it. If that happens, waitpid will return -1.
     retcode = waitpid(PtyIOStreams[pty].childPID, &status, WNOHANG);
     if (retcode == 0) {
         // Give process a second to quit
```

Comment:
I know this wasn't added in this PR (but the previous one which I didn't review), but the one second seems rather arbitrary (if usually enough for a process to quit). Does it make sense to have this waiting time configurable, or at least definable?

Reply:
I admit it's arbitary, but I'm not sure I'd want to add a configuration option people might not understand.

In general, if an app hasn't quite within a second of being sent a `SIGTERM`, then it's a fairly badly behaved program.

One option would be to wait less long before going back to see if the program has stopped, maybe waiting 0.1 seconds 10 times, so GAP can continue faster.

Reply:
Now you bring it up, I'm tempted to make the timeout much longer (say 10 seconds), and check more often. That should deal with almost every case in practice, and make sure GAP keeps working for particularly badly behaved subproblems.

We would never have found this if there wasn't buggy copies of `rev` floating around which swallow all signals and enter an infinite loop.